lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -6352,17 +6357,21 @@ void Differentiation::run() {
   // A global differentiation context.
   ADContext context(*this);
 
-  // Handle all the instructions and attributes in the module that trigger
-  // differentiation.
+  // Use a temporary list to store `@differentiable` attributes and invokers
+  // for deterministic iteration order.
+  SmallVector<std::pair<SILDifferentiableAttr *, DifferentiationInvoker>, 32>
+      invokerList;
```

Comment:
Have you thought about using `llvm::MapVector`?

Reply:
I hadn't. Changed to `llvm::SmallMapVector` in https://github.com/apple/swift/pull/25268/commits/cbb9c7089aa394ece74c5998d33b3b6301ff0616.